src/main/java/com/formulasearchengine/mathmlsim/similarity/SubTreeComparison.java

```diff
@@ -0,0 +1,174 @@
+                continue;
+            }
+            // go deeper in the comp. tree but hold the ref tree
+            if (findSimilarities(refTree, compChild, similarities, true, onlyOperators)) {
+                return true;
+            }
+        }
+
+        if (!holdRefTree) {
+            // go deeper in the reference tree
+            for (MathNode refChild : refTree.getChildren()) {
+                if (onlyOperators && refChild.isLeaf()) {
+                    continue;
+                }
+                findSimilarities(refChild, comTree, similarities, false, onlyOperators);
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Are aTree and bTree identical subtrees? If the root node is equal,
+     * all subsequent children will be compared.
+     *
+     * @param aTree first MathNode tree
+     * @param bTree second MathNode tree
+     * @return true - if both trees are identical subtrees, false otherwise
+     */
+    boolean isIdenticalTree(MathNode aTree, MathNode bTree) {
+        // first check if they have the same number of children
+        if (aTree.equals(bTree) && aTree.getChildren().size() == bTree.getChildren().size()) {
+            if (aTree.isOrderSensitive()) {
+                // all children order sensitive
+                for (int i = 0; i < aTree.getChildren().size(); i++) {
+                    if (!isIdenticalTree(aTree.getChildren().get(i), bTree.getChildren().get(i))) {
+                        return false;
+                    }
+                }
+            } else {
+                // order insensitive
+                List<MathNode> bChildren = new ArrayList<>(bTree.getChildren());
+                OUTER:
+                for (MathNode aChild : aTree.getChildren()) {
+                    for (MathNode bChild : filterSameChildren(aChild, bChildren)) {
+                        if (isIdenticalTree(aChild, bChild)) {
+                            // found an identical child
+                            bChildren.remove(bChild);
+                            continue OUTER;
+                        }
+                    }
+                    // aChild is missing in bChildren
+                    return false;
+                }
+            }
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * Filter for similar nodes from a list. Only the node itself will be compared.
+     * We will not look at their respective children.
+     * <br/>
+     * This method is used when we want to compare children from different nodes
+     * out of order.
+     *
+     * @param searchNode node we want to find.
+     * @param list       list we want to search.
+     * @return new list with the same node we search for.
+     */
+    List<MathNode> filterSameChildren(MathNode searchNode, List<MathNode> list) {
+        return list.stream().filter(searchNode::equals).collect(Collectors.toList());
+    }
+
+    /**
+     * This method is still in testing.
```

Comment:
This method was fine in the mean time, but the other assessment method were not. I will open an issue.